rust/automerge/src/text_value.rs

```diff
@@ -1,124 +1,122 @@
-use core::fmt::Debug;
-
 use crate::sequence_tree::SequenceTree;
+use cfg_if::cfg_if;
+use core::fmt::Debug;
 
-#[cfg(not(any(target_family = "wasm", feature = "utf8-indexing")))]
-#[derive(Clone, PartialEq, Default)]
-pub struct TextValue(SequenceTree<char>);
-
-#[cfg(target_family = "wasm")]
-#[derive(Clone, PartialEq, Default)]
-pub struct TextValue(SequenceTree<u16>);
-
-#[cfg(feature = "utf8-indexing")]
-#[derive(Clone, PartialEq, Default)]
-pub struct TextValue(SequenceTree<u8>);
-
-#[cfg(not(any(target_family = "wasm", feature = "utf8-indexing")))]
```

Comment:
Switching from this style to using `cfg-if` doesn't seem worth it to me. We're taking on an extra dependency for a reasonably small increase in readability IMO. As I understand it the difference is really that we would go from `#[cfg(feature = <...>)]` to the nested if chains below right?

Reply:
100% aligned with your comment. Basically, I do not want to introduce any if-else, or another dependency for solving that. 

It’s unclear whether this is a limitation of Rust or an intentional design choice, but in Rust, features are not mutually exclusive. This means you can’t easily express conditions like <if featureA and !featureB>. There's no direct syntax for writing this, aside from using:
- When there is a conflict, [choose one feature over another](https://doc.rust-lang.org/cargo/reference/features-examples.html#feature-precedence). The [cfg-if](https://crates.io/crates/cfg-if) package can help with writing more complex `cfg` expressions. **It is the recommended, and followed approach**
- Split the functionality into separate packages.
- Architect the code to allow the features to be enabled concurrently, and use runtime options to control which is used. For example, use a config file, command-line argument, or environment variable to choose which behavior to enable. **Not a valid option**

👉🏻 Reference from official Rust doc [[Mutually exclusive features]](https://doc.rust-lang.org/cargo/reference/features.html#mutually-exclusive-features)